cmd/esc/cli/client/client.go

```diff
@@ -407,26 +435,32 @@ func (pc *client) ListEnvironments(
 	}
 
 	var resp ListEnvironmentsResponse
-	err := pc.restCall(ctx, http.MethodGet, "/api/preview/environments", queryObj, nil, &resp)
+	err := pc.restCall(ctx, http.MethodGet, "/api/esc/environments", queryObj, nil, &resp)
 	if err != nil {
 		return nil, "", err
 	}
 	return resp.Environments, resp.NextToken, nil
 }
 
+// Deprecated
 func (pc *client) CreateEnvironment(ctx context.Context, orgName, envName string) error {
-	path := fmt.Sprintf("/api/preview/environments/%v/%v", orgName, envName)
+	return pc.CreateEnvironmentWithProject(ctx, orgName, DefaultProject, envName)
+}
+
+func (pc *client) CreateEnvironmentWithProject(ctx context.Context, orgName, projectName, envName string) error {
+	path := fmt.Sprintf("/api/esc/environments/%v/%v/%v", orgName, projectName, envName)
```

Comment:
Do we plan on renaming this after the pulumi CLI update, once the deprecated function can be removed or are we always going to stick with `CreateEnvironmentWithProject`? If the latter, maybe we can call this something like `CreateProjectEnvironment` or `CreateEnvironmentInProject` since the environment will belong to the project

Reply:
I would prefer removing the deprecated function and renaming, if we're okay with breaking the esc package api

Reply:
yup that sounds right